samples/nrf9160/gps/README.rst

```diff
@@ -1,19 +1,26 @@
 .. _gps_with_supl_support_sample:
 
-nRF9160: GPS sockets and SUPL client library
-############################################
+nRF9160: GPS with SUPL client library
+#####################################
 
 .. contents::
    :local:
    :depth: 2
 
-The GPS sample demonstrates how to retrieve `GPS`_ data.
-If Secure User-Plane Location (SUPL) support is enabled, it also shows how to improve the GPS fix accuracy and fix speed with `A-GPS`_ data from a SUPL server.
+The GPS sample demonstrates how to retrieve `GNSS`_ data.
+If Secure User-Plane Location (SUPL) support is enabled, it also shows how to improve the fix accuracy and fix speed with `A-GPS`_ data from a SUPL server.
 See :ref:`supl_client` for information on enabling SUPL support for the sample.
 
-The sample first creates a `GNSS`_ socket.
-Then it reads the GPS data from the socket and parses the received frames to interpret the frame contents.
-See :ref:`nrfxlib:gnss_extension` for more information.
+The sample first initializes the `GNSS`_ interface.
+Then it handles events from the interface, reads the associated data and outputs information to the console.
+Because `NMEA`_ data needs to be read as soon as an NMEA event is received, a :ref:`Zephyr message queue <zephyr:message_queues_v2>` is used for buffering the NMEA strings.
+The event handler function reads the received NMEA strings and puts those into the message queue.
+The consumer loop reads from the queue and outputs the strings to the console.
+
+When SUPL support is enabled, a :ref:`Zephyr workqueue <zephyr:workqueues_v2>` is used for downloading the `A-GPS`_ data.
```

Comment:
Leave out the A-GPS link as we have it already on line 11. Also consider getting rid of the same link under "Using the sample with..." just before the Testing section.